Added explanations of Git LFS and VFS in new .md file (#586)
microsoft/code-with-engineering-playbook: source-control/git-guidance/git-lfs-and-vfs.md

The way it works in short, is that a placeholder file is stored in the repo with information for the LFS system. It looks something like this:

```shell
version https://git-lfs.github.com/spec/v1
oid a747cfbbef63fc0a3f5ffca332ae486ee7bf77c1d1b9b2de02e261ef97d085fe
size 4923023
```

The actual file is stored in a separate storage. This way Git will track changes in this placeholder file, not the large file. The combination of using Git and Git LFS will hide this from the developer though. You will just work with the repository and files as before.

When working with these large files yourself, you'll still see the git history grown on your own machine, as git will still start tracking these large files locally. But when you clone the repo, the history is actually pretty small. So it's beneficial for others not working directly on the large files.

### Benefits of Git LFS

* Uses the end to end Git workflow for all files
* Git LFS supports file locking to avoid conflicts for undiffable assets
* Git LFS is fully supported in Azure DevOps Services

> ### Limitations of Git LFS

Reviewer: Are these really limitations or pros/cons?

Reviewer: We should mention that LFS is a paid service.

Author: @cmayomsft LFS is not a paid service, as far as I know - but please tell me if I'm wrong

Author: @cmayomsft good point, adjusted pros/cons!

```

See the [Git LFS installation instructions](https://github.com/git-lfs/git-lfs/wiki/Installation) for installation on other systems

Track .mp4 files with Git LFS

```bash
git lfs track '*.mp4'
```

Update the `.gitattributes` file listing the files and patterns to track

```bash
*.mp4 filter=lfs diff=lfs merge=lfs -text
docs/images/* filter=lfs diff=lfs merge=lfs -text
```

List all patterns tracked

```bash
git lfs track
```

List all files tracked

```bash
git lfs ls-files
```

Download files to your working directory

```bash
git lfs pull
git lfs pull --include="path/to/file"
```

## VFS for Git

Imagine a large repository with multiple projects in it, for instance for all kinds of features. A developer in a specific team is working on just one feature. He doesn't require *all* files of the repo to work on that one feature. But normally with Git, you will clone *all* files in the repo.

VFS for Git (or Virtual File System for Git) solves this problem, as it will only download what you need to your local machine. But if you look in the file system, e.g. with Windows Explorer, it will show all the folders and files including the correct file sizes.

> When you use VFS for Git for a repository a process is started that does all the work to hide that downloading.

Reviewer 2: ```suggestion
```

Reviewer 2: I don't think this sentence is needed - and it is a little confusing as it suggests that it is in fact downloading but the download is hidden to the user (or is that what really happens?)

The necessary installers can be found at [https://github.com/Microsoft/VFSForGit/releases](https://github.com/Microsoft/VFSForGit/releases)

On the releases page you'll find two important downloads:

* Git 2.28.0.0 installer, which is a requirement for running VFS for Git. This is not the same as the standard Git for Windows install!
* SetupGVFS installer.

Download those files and install them on your machine.

To be able to use VFS for Git for a repository, a `.gitattributes` file needs to be added to the repo with this line in it:

```shell
* -text
```

To clone a repository to your machine using VFS for Git you use `gvfs` instead of `git` like so:

```shell
gvfs clone [URL] [dir]
```

Once this is done, you have a folder which contains a `src` folder which contains the contents of the repository. This is done because of a practice to put all outputs of build systems outside of this tree. This makes it easier to manage `.gitignore` files and to keep Git performant with lots of files.

For working with the repository you just use Git commands as before.

> One thing to know is how to remove a VFS for Git repository from your machine. You also want to make sure the VFS process is stopped. This can be done by executing this command inside the main folder:

Reviewer 2: ```suggestion
To remove a VFS for Git repository from your machine, make sure the VFS process is stopped and execute this command from the main folder:
```
